src/validation.cpp

```diff
@@ -81,12 +81,14 @@ ChainstateManager g_chainman;
 
 CChainState& ChainstateActive()
 {
+    LOCK(::cs_main);
     assert(g_chainman.m_active_chainstate);
     return *g_chainman.m_active_chainstate;
 }
 
 CChain& ChainActive()
 {
+    LOCK(::cs_main);
```

Comment:
c9017ce

Is this LOCK really necessary given we call `::ChainstateActive()` just after ? Doesn't hurt because we use recursive_mutex but still..

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17737#discussion_r395205730

> Is this LOCK really necessary given we call `::ChainstateActive()` just after ? Doesn't hurt because we use recursive_mutex but still..

Agree it's definitely not necessary (but harmless)